pkg/operator/mon/mon.go

```diff
@@ -199,14 +208,19 @@ func (c *Cluster) initMonServices(size int) ([]*monConfig, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to create mon service. %+v", err)
 		}
-		m.PublicIP = serviceIP
-		c.clusterInfo.Monitors[m.Name] = mon.ToCephMon(m.Name, m.PublicIP)
+		if !c.HostNetwork {
+			m.PublicIP = serviceIP
+			c.clusterInfo.Monitors[m.Name] = mon.ToCephMon(m.Name, m.PublicIP)
+		}
 	}
 
```

Comment:
could you assign pods here? Then you could call `saveMonConfig` for both hostNetwork true or false